test/testdialog.c

```diff
@@ -0,0 +1,148 @@
+/*
+  Copyright (C) 1997-2023 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely.
+*/
+/* Sample program:  Create open and save dialogs. */
+
+#include <SDL3/SDL.h>
+#include <SDL3/SDL_main.h>
+#include <SDL3/SDL_test.h>
+
+const SDL_DialogFileFilter filters[4] = {
+    { "All files", "*" },
+    { "JPG images", "jpg;jpeg" },
+    { "PNG images", "png" },
+    { NULL, NULL }
+};
+
+static void callback(void* userdata, const char* const* files, int filter) {
```

Comment:
This must be marked as SDLCALL: `static void SDLCALL callback(void* userdata, ..............`